contracts/hydro/src/contract.rs

```diff
@@ -676,135 +675,210 @@ fn vote(
+                    lock_entry.funds.denom
+                ));
 
-        let scaled_shares = get_lock_time_weighted_shares(round_end, lock_entry, lock_epoch_length);
+                // skip this lock entry, since the locked shares do not belong to a validator that we want to take into account
+                continue;
+            }
+            let validator = validator_result.unwrap();
 
-        // add the shares to the map
-        let shares = match time_weighted_shares_map.get(&validator) {
-            Some(shares) => *shares,
-            None => Decimal::zero(),
-        };
-        let new_shares = shares.checked_add(Decimal::from_ratio(scaled_shares, Uint128::one()))?;
+            let scaled_shares =
+                get_lock_time_weighted_shares(round_end, lock_entry, lock_epoch_length);
 
-        // insert the shares into the time_weigted_shares_map
-        time_weighted_shares_map.insert(validator.clone(), new_shares);
-    }
+            // add the shares to the map
+            let shares = match time_weighted_shares_map.get(&validator) {
+                Some(shares) => *shares,
+                None => Decimal::zero(),
+            };
+            let new_shares =
+                shares.checked_add(Decimal::from_ratio(scaled_shares, Uint128::one()))?;
 
-    // if the user doesn't have any shares that give voting power, we don't need to do anything
-    if time_weighted_shares_map.is_empty() {
-        return Ok(response);
-    }
+            // insert the shares into the time_weigted_shares_map
+            time_weighted_shares_map.insert(validator.clone(), new_shares);
 
-    // update the proposal's power with the new shares
-    for (validator, num_shares) in time_weighted_shares_map.iter() {
-        // add the validator shares to the proposal
-        add_validator_shares_to_proposal(
-            deps.storage,
-            round_id,
-            proposal_id,
-            validator.to_string(),
-            *num_shares,
-        )?;
-    }
+            // if the user doesn't have any shares that give voting power, we don't need to do anything
+            if time_weighted_shares_map.is_empty() {
+                return Ok(response);
+            }
 
-    // update the proposal in the proposal map, as well as the props by score map
-    update_proposal_and_props_by_score_maps(deps.storage, round_id, tranche_id, proposal_id)?;
+            // update the proposal's power with the new shares
+            for (validator, num_shares) in time_weighted_shares_map.iter() {
+                // add the validator shares to the proposal
+                add_validator_shares_to_proposal(
+                    deps.storage,
+                    round_id,
+                    proposal_id,
+                    validator.to_string(),
+                    *num_shares,
+                )?;
+            }
```

Comment:
same, there will only be 1 entry

Reply:
Done in https://github.com/informalsystems/hydro/pull/157/commits/21e74b10f1675c18a3fa0df0beb1d1c15f0458b7